### ISSUE TITLE
Moving surface slice intersection is not visible

### ISSUE THREAD

**Issue by cpinter** [SlicerProstate/SlicerProstate#9]
The surface of the moving image after the registration with DistanceMapBasedRegistration (the blue model) is not visible by default on the 2D slices after registration. Although it can be made visible by going to the Models module and turning on slice intersections for that model, but it would make it easier to see the result immediately if it was enabled by default.

**Comment by fedorov** [comment]
@cpinter I will commit the fix soon. I am also taking this as an opportunity to replace print() with logging(). Do you know what I need to do to initialize logging properly? When I do logging.info() in my module, I get messages like

```
[INFO][Stream] 11.02.2016 13:57:52 [] (unknown:0) - Cropped volume pushed
```

What do I need to do to make it show my module name instead of "unknown"? I will try to find this now in parallel.

**Comment by cpinter** [comment]
Thanks Andriy! I don't know what is needed to better identify the place of the message with python logging. I simply prefixed each message with the function name, but it's not a very good solution.

**Comment by lassoan** [comment]
logging.debug logs file name and line number properly. Log levels that print message on the console (info, warning, error) produce a log entry because all python outputs are logged. We could enable adding an additional log entry with file name and line numbers but that would mean two entries for each info, warning, or error message. Do you think we should enable this duplicate logging?

**Comment by fedorov** [comment]
When I look at the log, some messages (even info messages) are prefixed by the module where they are coming from, for example

```
[INFO][VTK] 11.02.2016 13:57:38 [vtkMRMLVolumeArchetypeStorageNode (0x13e9b4570)]
```

Do you know how something similar could be done so that it is known what is the module that produced a log message?

I agree, duplicate messages don't look nice. I wonder if there is something else we can do to add more detail. Maybe it can even be as simple as this example from https://docs.python.org/2/library/logging.html#logger-objects:

```
FORMAT = '%(asctime)-15s %(clientip)s %(user)-8s %(message)s'
logging.basicConfig(format=FORMAT)
d = {'clientip': '192.168.0.1', 'user': 'fbloggs'}
logger = logging.getLogger('tcpserver')
logger.warning('Protocol problem: %s', 'connection reset', extra=d)
```

which produces

```
2006-02-08 22:20:02,165 192.168.0.1 fbloggs  Protocol problem: connection reset
```

and extra could be defined at the parent class, or automatically from the class name?

**Comment by lassoan** [comment]
This already works for debug. If you enter logging.debug("this is a debug message") on the console then in the log you'll see this:

```
[DEBUG][Python] 11.02.2016 14:29:28 [Python] (<console>:1) - this is a debug message
[DEBUG][Qt] 11.02.2016 14:29:28 [] (unknown:0) - Python console user input: logging.debug("this is a debug message")
```

(if you log from a file then the filename appears instead of <console>)

If we enable full logging for higher level messages (info, warning, debug) then we get this:

```
[WARNING][Python] 11.02.2016 14:28:29 [Python] (<console>:1) - this is a warning message
[DEBUG][Qt] 11.02.2016 14:28:29 [] (unknown:0) - Python console user input: logging.warning("this is a warning message")
[CRITICAL][Stream] 11.02.2016 14:28:29 [] (unknown:0) - this is a warning message
```

The first line contain all the details and proper log level. The second line is OK, it's just shows what we entered on the console. The third line is basically a duplicate of the error that is logged in the first line.

Would you prefer to have enable proper logging at every level?

You can try it by commenting out this line in the file `...Python/slicer/__init__.py`:

applicationLogHandler.addFilter(_LogReverseLevelFilter(logging.INFO)) line

**Comment by lassoan** [comment]
We already have the full logging capability at all level, it's just disabled for info, warning, error to prevent duplicate logs. Probably duplicate log entries are less evil than missing file name & line number, so it may be a good idea to revisit the decision of suppressing the detailed log entry.

**Comment by fedorov** [comment]
yes, when I comment that line, I get [INFO][Python] with the filenames and line numbers, and a duplicate [INFO][Stream], as you said. It is a bit tricky, since it is blocks of logging messages that are interleaved, [Stream] does not immediately follow [Python]. But maybe it is ok, since those are for developers only, and they can always be sorted by time.

What is the meaning of [Stream] log? Is there a situation where we need it?

**Comment by lassoan** [comment]
Stream log logs anything that is displayed on the python console. Some messages are simply printed on the console and not using logging macros. Those can only be captured if we write all python console output to the log.

**Comment by fedorov** [comment]
I see. So yes, I agree, not adding that filter should be helpful in utilizing the logging messages better for troubleshooting.
